src/shared/particles/base_particles.hpp

```diff
@@ -35,400 +35,369 @@
+{
+    registerVariable(variable_addrs, variable_name);
+    for (size_t i = 0; i != real_particles_bound_; ++i)
     {
-        constexpr int type_index = DataTypeIndex<DataType>::value;
-        bool isRegistered = false;
-
-        for (auto& variable : std::get<type_index>(all_global_data_))
-        {
-            if (variable->getName() == variable_name)
-            {
-                isRegistered = true;
-                return variable->getValue();
-            }
-        }
+        variable_addrs[i] = initialization(i); // Here, lambda function is applied for initialization.
+    }
+}
+//=================================================================================================//
+template <typename DataType>
+DataType *BaseParticles::registerGlobalVariable(const std::string &variable_name, DataType initial_value)
+{
+    GlobalVariable<DataType> *variable = findVariableByName<DataType>(all_global_variables_, variable_name);
 
-        if (!isRegistered)
-        {
-            UniquePtrsKeeper<GlobalVariable<DataType>>& container = std::get<type_index>(all_global_data_ptr_);
-            GlobalVariable<DataType> *contained_data = container.template createPtr<GlobalVariable<DataType>>(variable_name, initial_value);
-            std::get<type_index>(all_global_data_).push_back(contained_data);
-            return contained_data->getValue();
-        }
+    return variable != nullptr
+               ? variable->ValueAddress()
+               : addVariableToAssemble<DataType>(all_global_variables_,
+                                                 all_global_variable_ptrs_, variable_name, initial_value)
+                     ->ValueAddress();
+}
+//=================================================================================================//
+template <typename DataType>
+DataType *BaseParticles::getGlobalVariableByName(const std::string &variable_name)
+{
+    GlobalVariable<DataType> *variable = findVariableByName<DataType>(all_global_variables_, variable_name);
 
-        return nullptr;
-    }
-     //=================================================================================================//
-    template <typename DataType>
-    DataType *BaseParticles::getGlobalVariableByName(const std::string &variable_name)
+    if (variable != nullptr)
     {
-        constexpr int type_index = DataTypeIndex<DataType>::value;
-        bool isRegistered = false;
+        return variable->ValueAddress();
+    }
 
-        for (auto& variable : std::get<type_index>(all_global_data_))
-        {
-            if (variable->getName() == variable_name)
-            {
-                isRegistered = true;
-                return variable->getValue();
-            }
-        }
+    std::cout << "\nError: the variable '" << variable_name << "' is not registered!\n";
+    std::cout << __FILE__ << ':' << __LINE__ << std::endl;
+    exit(1);
```

Comment:
```suggestion
```

Reply:
we need to exit the code if the error happens.

Reply:
See updated suggestion and comment above